### ISSUE TITLE
[BUG] etcdctl with high UID causes docker error during image extraction due to failed user id remapping

### ISSUE THREAD

**Issue by BassMonkey** [rancher/rancher#39588]
**Rancher Server Setup**
- OS: Ubuntu 22.04
- Docker version: 20.10.21
- Rancher version: 2.6.9
- Installation option: Docker install
- Proxy/Cert Details: no proxy, auto-generated cert

**Information about the Cluster**
not relevant, install already failing

**User Information**
- What is the role of the user logged in?
admin

**Describe the bug**
I am using user namespace remapping (userns-remap), which is the recommended setup to ensure better isolation:
https://docs.docker.com/engine/security/userns-remap/

With that setup pulling the current rancher image docker errors out during extraction of the image due to a failed user id remapping:
```
ubuntu@ubuntu-rpi-4-2:~$ docker run -d --restart=unless-stopped -p 80:80 -p 443:443 --userns=host --privileged rancher/rancher:latest
[...]
docker: failed to register layer: Error processing tar file(exit status 1): Container ID 114762 cannot be mapped to a host ID.
```

The cause is the file "/usr/bin/etcdctl" in the image, which has an unusually high UID. That's likely an issue in the image build pipeline:
```
ubuntu@ubuntu-rpi-4-2:~$ sudo docker exec -it acd818b6d0fc /bin/bash
acd818b6d0fc:/var/lib/rancher # cd /
acd818b6d0fc:/ # find / \( -uid 114762 \)  -ls 2>/dev/null
   282361  16320 -rwxr-xr-x   1 114762   114762   16711680 Oct 15  2021 /usr/bin/etcdctl
```

This article describes the issue very well:
https://circleci.com/docs/high-uid-error/

To fix it you need to make sure that during build no files have a UID higher than 65535.

**To Reproduce**
- Start the docker daemon with userns-remap enabled
- Pull or run rancher docker image v2.6.9

**Result**
Image extraction fails with the following error:
`docker: failed to register layer: Error processing tar file(exit status 1): Container ID 114762 cannot be mapped to a host ID.`

**Expected Result**
Docker image pulled and extracted successfully.

**Additional context**

Full Docker config:
ubuntu@ubuntu-rpi-4-2:~$ cat /etc/docker/daemon.json 
{
  "storage-driver": "overlay2",
  "graph": "/var/lib/docker",
  "log-driver": "json-file",
  "log-opts": {
    "max-size": "10m",
    "max-file": "2"
  },
  "debug": false,
  "userns-remap": "1000:1000"   <<<<<<< this line means that user namespaces are used by docker
}

Full output when pulling rancher image:
```
ubuntu@ubuntu-rpi-4-2:~$ docker run -d --restart=unless-stopped -p 80:80 -p 443:443 --userns=host --privileged rancher/rancher:latest
Unable to find image 'rancher/rancher:latest' locally
latest: Pulling from rancher/rancher
da348f545abb: Pull complete 
10819158b6b3: Pull complete 
eabaac58afa2: Pull complete 
9130a3911e30: Pull complete 
c7f7ca81f962: Pull complete 
b8ad8dcdea91: Pull complete 
5cf798afd722: Pull complete 
307f58f62784: Pull complete 
276ce0ac207c: Pull complete 
f238be7c4e4c: Pull complete 
f9d7d97c66d6: Pull complete 
8258f1fe24ec: Extracting [==================================================>]  9.496MB/9.496MB
e446d754f3c6: Download complete 
c2c732a6f76b: Download complete 
28911ba28af6: Download complete 
0beee4bdc8d6: Download complete 
1b4e09765e22: Download complete 
3302d9e0b6f4: Download complete 
718484cdfb43: Download complete 
65bcb8ad7931: Download complete 
docker: failed to register layer: Error processing tar file(exit status 1): Container ID 114762 cannot be mapped to a host ID.
See 'docker run --help'.
```

SURE-5761

**Comment by snasovich** [comment]
At a glance it looks like we're extracting etcdctl here and that's where we want to ensure UID is under 65K: https://github.com/rancher/rancher/blob/9278cf152f9b98ae3c543c0f956af2ae6ab91368/package/Dockerfile#L157

**Comment by sowmyav27** [comment]
Reopening for automated tests

**Comment by Sahota1225** [comment]
justification of why test are not added is in the PR

**Comment by bfbachmann** [comment]
Copying some of [the PR](https://github.com/rancher/rancher/pull/41165) description here for convenience:

> This change doesn't really change anything about Rancher that is really testable. It just changes the owner of a file copied during the process of building the Rancher image. In this case, I'd say the automated test is essentially the build process itself.

**Comment by markusewalker** [comment]
**QA TEST PLAN**

Scenarios
|Scenario|Test Case|
|---|---|
|1|Attempt to pull and run Rancher image with a system that has `userns-remap` configured in dockerd|

**Comment by markusewalker** [comment]
Reproduced this behavior on `v2.7.2`. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.7.2`

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|Attempt to pull and run Rancher image with a system that has `userns-remap` configured in dockerd|:x:|

**REPRO STEPS**

**Scenario 1**
1. On a client machine, ran the following commands as the root user to add a test user and add to the `docker` group:
```
$ useradd -m testuser
# usermod -aG docker testuser
```
2. Validated that a respective uid and gid exist:
```
$ id testuser
uid=1001(testuser) gid=1001(testuser) groups=1001(testuser),999(docker)
```
3. Ensured that docker is not started and ran command `dockerd --userns-remap="testuser:testuser"`.
4. Reproduced the issue when attempting to run a Rancher 2.7.2:
```
$ docker run -d -e "CATTLE_BOOTSTRAP_PASSWORD=…" --restart unless-stopped -p 80:80 -p 443:443 --userns=host --privileged "rancher/rancher:v2.7.2"
Unable to find image 'rancher/rancher:v2.7.2' locally
v2.7.2: Pulling from rancher/rancher
41fdb87eda23: Pull complete 
6ab05bf9a947: Pull complete 
f22b93c683d3: Pull complete 
6ca95739a99b: Pull complete 
677543b96838: Pull complete 
f9a72b67d821: Pull complete 
e778bc01834b: Pull complete 
016f3998a7b6: Pull complete 
cbfc612dd379: Pull complete 
f6b8646a8e97: Pull complete 
e8a9c9fbac9d: Pull complete 
d5640d2a01e7: Extracting [==================================================>]   10.4MB/10.4MB
79c4f912e9c1: Download complete 
20687e623d56: Download complete 
5bf6785e931d: Download complete 
03d93aadf9ca: Download complete 
08f5ddcd57b7: Download complete 
9143f11c8392: Download complete 
34a22cc63e61: Download complete 
9a0f94cdd348: Download complete 
docker: failed to register layer: Error processing tar file(exit status 1): Container ID 114762 cannot be mapped to a host ID.
```
Now that I have reproduced this issue, I will attempt to validate if this has been addressed in `v2.7-head`.

**Comment by markusewalker** [comment]
Validated that this has been addressed on `v2.7-head` https://github.com/rancher/rancher/commit/2efc8340e83d8f21bb0553f13e65e2b25a45821d. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/2efc8340e83d8f21bb0553f13e65e2b25a45821d

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|Attempt to pull and run Rancher image with a system that has `userns-remap` configured in dockerd|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. On a client machine, ran the following commands as the root user to add a test user and add to the `docker` group:
```
$ useradd -m testuser
$ usermod -aG docker testuser
```
2. Validated that a respective uid and gid exist:
```
$ id testuser
uid=1001(testuser) gid=1001(testuser) groups=1001(testuser),999(docker)
```
3. Ensured that docker is not started and ran command `dockerd --userns-remap="testuser:testuser"`.
4. Reproduced the issue when attempting to run a Rancher 2.7.2:
```
$ docker run -d -e "CATTLE_BOOTSTRAP_PASSWORD=..." --restart unless-stopped -p 80:80 -p 443:443 --userns=host --privileged "rancher/rancher:v2.7-head"
Unable to find image 'rancher/rancher:v2.7-head' locally
v2.7-head: Pulling from rancher/rancher
b09c5f2087db: Pull complete 
f6fc17dcd167: Pull complete 
f4b0fb4848a9: Pull complete 
bf5c5107715a: Pull complete 
1de62a0ad862: Pull complete 
15429cbe5a8c: Pull complete 
b97c7488a04a: Pull complete 
330ef1ffb904: Pull complete 
87bb244a2c9d: Pull complete 
a8084c5eff96: Pull complete 
3ef766724a8b: Pull complete 
95014249346a: Pull complete 
c5c39f6a066a: Pull complete 
34cf615d3c0b: Pull complete 
664a546cd512: Pull complete 
10686f0fe716: Pull complete 
d3fe30f2b6fd: Pull complete 
fcad6bc62311: Pull complete 
42411fb55228: Pull complete 
7e7521b0aca1: Pull complete 
Digest: sha256:dc0e6c77df2bdcb632eb7e57e4d0fa38cca95e3ece4f60b6b6267df25bf7eb7a
Status: Downloaded newer image for rancher/rancher:v2.7-head
60f0dc909a1b7d53d26dff4149852d48ea49f861df249810ef6e7d2851a3ab4a
```
5. Ran command `docker exec -it <running Rancher container> /bin/bash` and validated that the uid and gid of `etcdctl` is the same as the other tools in the `/usr/bin` directory.
